### PR TITLE
Fix issues with docker-compose and imporove deploy logic

### PR DESCRIPTION
**Pull request by sadeden**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -14,12 +14,11 @@ RUN apt-get update && apt-get install -qq -y libxml2 libxml2-dev git build-essen
 
 # Set an environment variable to store where the app is installed to inside
 # of the Docker image.
-ENV INSTALL_PATH /mmt
-RUN mkdir -p $INSTALL_PATH
+RUN mkdir -p /mnt
```
Comment (nikolai-b) [inline Dockerfile]: as we're using `/mnt` a few times should kept it as an `ENV`?
I thought `INSTALL_PATH` made sense?

File: Dockerfile [hunk 2]
```diff
@@ -14,12 +14,11 @@ RUN apt-get update && apt-get install -qq -y libxml2 libxml2-dev git build-essen
...
-RUN mkdir -p $INSTALL_PATH
+RUN mkdir -p /mnt 
 
 # This sets the context of where commands will be ran in and is documented
 # on Docker's website extensively.
-WORKDIR $INSTALL_PATH
+WORKDIR /mnt
 
 # Ensure gems are cached and only get updated when they change. This will
 # drastically increase build times when your gems do not change.
```
Comment (nikolai-b) [inline Dockerfile]: Should it not be
```
COPY Gemfile Gemfile.lock $APP_HOME
```
?

File: Dockerfile [hunk 3]
```diff
@@ -28,7 +27,7 @@ RUN bundle install
```
Comment (nikolai-b) [inline Dockerfile]: I think before we `RUN bundle install` we should have installed bundler `RUN gem install bundler -v 1.15.4`

File: docker-compose.yml [hunk 4]
```diff
@@ -1,39 +1,43 @@
...
-  volumes:
-    - .:/mmt
-  env_file:
-    - .mmt.env
+services:
+  postgres:
+    image: postgres:9.6.5
+    environment:
+      POSTGRES_USER: postgres
+      POSTGRES_PASSWORD:
```
Comment (nikolai-b) [inline docker-compose.yml]: Maybe we should set a password, I know it uses `trust` https://hub.docker.com/_/postgres/ but it sometimes makes life easier :woman_shrugging:

File: docker-compose.yml [hunk 5]
```diff
@@ -1,39 +1,43 @@
...
+  mmt:
+    build: .
+    depends_on:
+      - postgres
+        #    - redis
+    volumes:
+      - .:/mmt
+    entrypoint: docker/entrypoint.sh
+    ports:
+      - '3000:3000'
```
Comment (nikolai-b) [inline docker-compose.yml]: I'd put 
```
ENV PORT=80 # and maybe HOST=0.0.0.0
EXPOSE 80 # not really needed...
```
in the Dockerfile and then have
```
ports:
  - '3001:80'
```
here as if we use the Docerfile in prod we'd expect it to serve on port 80

File: docker-compose.yml [hunk 6]
```diff
@@ -1,39 +1,43 @@
...
+  #  volumes:
+  #    - mmt-redis:/var/lib/redis/data
+  
+  mmt:
+    build: .
+    depends_on:
+      - postgres
+        #    - redis
+    volumes:
+      - .:/mmt
```
Comment (nikolai-b) [inline docker-compose.yml]: If you add
```
      - bundle:/usr/local/bundle
```
here and a 
```
volumes:
  bundle:
  mmt-postgres:
```
at the bottom then you can install gems without rebuilding Docker I think.

File: Dockerfile (hunk 1 above)
Comment (sadeden) [reply Dockerfile]: Sure, I'll put this back, I really wanted to simplify everything while debugging. I should have put this back.

**Comment by ameba23** [comment]
it looks good but im still not getting it to run.  im not sure how to create an issue on your fork of mmt
can you post the commands you used to build docker image?  next step would be to add this to the readme
beetroot ~/mmt (feature/dockerize●)$ sudo docker-compose up
Starting mmt_postgres_1 ... 
Starting mmt_postgres_1 ... done
Starting mmt_mmt_1 ... 
Starting mmt_mmt_1 ... done
Attaching to mmt_postgres_1, mmt_mmt_1
postgres_1  | LOG:  database system was shut down at 2017-11-27 09:47:06 UTC
postgres_1  | LOG:  MultiXact member wraparound protections are now enabled
postgres_1  | LOG:  autovacuum launcher started
postgres_1  | LOG:  database system is ready to accept connections
mmt_1       | bundler: failed to load command: rake (/usr/local/bundle/bin/rake)
mmt_1       | bundler: failed to load command: rake (/usr/local/bundle/bin/rake)
mmt_1       | Bundler::GemNotFound: Could not find rake-12.1.0 in any of the sources
mmt_1       |   /usr/local/lib/ruby/gems/2.4.0/gems/bundler-1.16.0/lib/bundler/spec_set.rb:88:in block in materialize'
mmt_1       |   /usr/local/lib/ruby/gems/2.4.0/gems/bundler-1.16.0/lib/bundler/spec_set.rb:82:in map!'
mmt_1       |   /usr/local/lib/ruby/gems/2.4.0/gems/bundler-1.16.0/lib/bundler/spec_set.rb:82:in materialize'
mmt_1       |   /usr/local/lib/ruby/gems/2.4.0/gems/bundler-1.16.0/lib/bundler/definition.rb:170:in specs'
mmt_1       |   /usr/local/lib/ruby/gems/2.4.0/gems/bundler-1.16.0/lib/bundler/definition.rb:237:in specs_for'
mmt_1       |   /usr/local/lib/ruby/gems/2.4.0/gems/bundler-1.16.0/lib/bundler/definition.rb:226:in requested_specs'
mmt_1       |   /usr/local/lib/ruby/gems/2.4.0/gems/bundler-1.16.0/lib/bundler/runtime.rb:108:in block in definition_method'
mmt_1       |   /usr/local/lib/ruby/gems/2.4.0/gems/bundler-1.16.0/lib/bundler/runtime.rb:20:in setup'
mmt_1       |   /usr/local/lib/ruby/gems/2.4.0/gems/bundler-1.16.0/lib/bundler.rb:107:in setup'
mmt_1       |   /usr/local/lib/ruby/gems/2.4.0/gems/bundler-1.16.0/lib/bundler/setup.rb:20:in <top (required)>'
mmt_1       |   /usr/local/lib/ruby/site_ruby/2.4.0/rubygems/core_ext/kernel_require.rb:59:in require'
mmt_1       |   /usr/local/lib/ruby/site_ruby/2.4.0/rubygems/core_ext/kernel_require.rb:59:in require'
mmt_mmt_1 exited with code 1
and if i add rake to the gemfile, and rebuild the docker image, i still get this error

**Comment by ameba23** [comment]
aha sadeden just explained.  

i was using wrong commands to build docker image

**Pull request merged**